lib/datamodel/graph.js

```diff
@@ -0,0 +1,79 @@
+/** Class that signifies a graph */
+class Graph {
+    /**
+      * The underlying graph object. This is an object with nodes and edges.
+      * The class contains an adjacency list property called graph that represents the edges
+      * between each of the nodes. Each edge can have attributes such as weight (which 
+      * by default is 10) and whether or not the edge is directed.
+      *
+      * */ 
```

Comment:
Add this comment directly above the `this.graph` frield. Also indicate that types: `Map<String, Map<String, Edge>>`